### ISSUE TITLE
forward references (strings) are allowed in invalid operations

### ISSUE THREAD

**Issue by KotlinIsland** [DetachHead/basedpyright#243]
```py
from typing import TypeAlias

yapyapyap: TypeAlias = "str" | int  # error: erm, actually, this is an error, soooo
a: TypeAlias = "str"
b: TypeAlias = a | None  # but this is okay, because that makes sense
```

```py
from typing import TypeAlias

A: TypeAlias = "int"

_ = A.as_integer_ratio(1)
```

**Comment by DetachHead** [comment]
if this isn't an issue when using the python 3.12 `type` keyword, then this will probably be low priority after #244 is done

**Comment by KotlinIsland** [comment]
I think it is if you ever want to resolve it
